extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go

```diff
@@ -68,123 +73,130 @@ func Responder(i *memif.Interface) error {
+			gopacket.SerializeLayers(buf, opts, &rEth, &rArp)
+			// write packet to shared memory
+			txq0.WritePacket(buf.Bytes())
+		}
+		if etherLayer.(*layers.Ethernet).EthernetType == layers.EthernetTypeIPv4 {
+			ipLayer := gopkt.Layer(layers.LayerTypeIPv4)
+			if ipLayer == nil {
+				fmt.Println("Missing IPv4 layer.")
 
-				// Set up buffer and options for serialization.
-				buf := gopacket.NewSerializeBuffer()
-				opts := gopacket.SerializeOptions{
-					FixLengths:       true,
-					ComputeChecksums: true,
-				}
-				gopacket.SerializeLayers(buf, opts, &ethResp, &ipv4Resp, &icmpResp,
-					gopacket.Payload(icmp.Payload))
-				// write packet to shared memory
-				txq0.WritePacket(buf.Bytes())
 			}
+			ipv4, _ := ipLayer.(*layers.IPv4)
+			if ipv4.Protocol != layers.IPProtocolICMPv4 {
+				fmt.Println("Not ICMPv4 protocol.")
+			}
+			icmpLayer := gopkt.Layer(layers.LayerTypeICMPv4)
+			if icmpLayer == nil {
+				fmt.Println("Missing ICMPv4 layer.")
+			}
+			icmp, _ := icmpLayer.(*layers.ICMPv4)
+			if icmp.TypeCode.Type() != layers.ICMPv4TypeEchoRequest {
+				fmt.Println("Not ICMPv4 echo request.")
+			}
+			//fmt.Println("Received an ICMPv4 echo request.")
 
-		}
-		return nil
+			// Build packet layers.
+			ethResp := layers.Ethernet{
+				DstMAC: net.HardwareAddr{0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa},
+				//DstMAC: net.HardwareAddr{0x02, 0xfe, 0xa8, 0x77, 0xaf, 0x20},
+				SrcMAC: []byte(net.HardwareAddr{0x02, 0xfe, 0x08, 0x88, 0x45, 0x7f}),
 
+				EthernetType: layers.EthernetTypeIPv4,
+			}
+			ipv4Resp := layers.IPv4{
+				Version:    4,
+				IHL:        5,
+				TOS:        0,
+				Id:         0,
+				Flags:      0,
+				FragOffset: 0,
+				TTL:        255,
+				Protocol:   layers.IPProtocolICMPv4,
+				SrcIP:      []byte("\xc0\xa8\x01\x01"),
+				DstIP:      []byte("\xc0\xa8\x01\x02"),
+			}
+			icmpResp := layers.ICMPv4{
+				TypeCode: layers.CreateICMPv4TypeCode(layers.ICMPv4TypeEchoReply, 0),
+				Id:       icmp.Id,
+				Seq:      icmp.Seq,
+			}
+
+			// Set up buffer and options for serialization.
+			buf := gopacket.NewSerializeBuffer()
+			opts := gopacket.SerializeOptions{
+				FixLengths:       true,
+				ComputeChecksums: true,
+			}
+			gopacket.SerializeLayers(buf, opts, &ethResp, &ipv4Resp, &icmpResp,
+				gopacket.Payload(icmp.Payload))
+			// write packet to shared memory
+			tx_bufs = append(tx_bufs, memif.MemifPacketBuffer{Buf: buf.Bytes(), Buflen: len(buf.Bytes())})
+
+		}
 	}
+	txq0.Tx_burst(tx_bufs)
+	return nil
 
 }
 func Connected(i *memif.Interface) error {
 	data, ok := i.GetPrivateData().(*interfaceData)
 	if !ok {
 		return fmt.Errorf("Invalid private data")
 	}
+	i.Pkt = make([]memif.MemifPacketBuffer, 64)
+	//pkt := make([]byte, 2048)
+
 	_ = data
 
 	// allocate packet buffer
-	pkt := make([]byte, 2048)
 	// get rx queue
 	rxq0, err := i.GetRxQueue(0)
 	_ = err
-
-	// read packet from shared memory
-	pktLen, err := rxq0.ReadPacket(pkt)
-	_, _ = err, pktLen
+	//TODO: fix Refill() to replace ReadPacket()
```

Comment:
We should probably address this before merging this

Reply:
Fixed: I just forgot to delete the comment. Refill() already replaced ReadPacket() in refilling queues.